images/pmd_cpd/pom.xml

```
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>sider</groupId>
+  <artifactId>runners</artifactId>
+  <version>1.0.0</version>
+  <dependencies>
+    <dependency>
+      <groupId>net.sourceforge.pmd</groupId>
+      <artifactId>pmd-java</artifactId>
```

[question] If we want to support all the languages, don't we need to change it as follows?

```suggestion
      <artifactId>pmd-dist</artifactId>
```

https://mvnrepository.com/artifact/net.sourceforge.pmd/pmd-dist

---

Good call!
The PMD CPD in pmd-java failed to analyze C/C++ source files. The pmd-dist package contains a support module for each language and enables us to analyze properly.
Fixed. commit 1475a277dacc5f6af25882246fa024115c1efa50